metricflow/model/data_warehouse_model_validator.py

```diff
@@ -41,86 +76,82 @@ class DataWarehouseValidationTask:
-
-        wrapped_cols = [DataWarehouseTaskBuilder._wrap_col(col, index) for index, col in enumerate(columns)]
-        columns_select = ", ".join(wrapped_cols)
-
-        query = DataWarehouseTaskBuilder.QUERY_TEMPLATE.format(
-            data_source=data_source_str, columns_select=columns_select, unique_id=id
+    def renderize(
+        sql_client: SqlClient, plan_converter: DataflowToSqlQueryPlanConverter, plan_id: str, nodes: FilterElementsNode
+    ):
+        """Generates a sql query plan and returns the rendered sql and execution_parameters"""
+        sql_plan = plan_converter.convert_to_sql_query_plan(
+            sql_engine_attributes=sql_client.sql_engine_attributes,
+            sql_query_plan_id=plan_id,
+            dataflow_plan_node=nodes,
         )
-        query += DataWarehouseTaskBuilder._where_clause_from_partitions(data_source=data_source)
-        return query
 
-    @classmethod
-    def _gen_query_and_params(
-        cls, data_source: DataSource, id: int, columns: List[str] = ["true"]
-    ) -> Tuple[str, SqlBindParameters]:
-        """A wrapping function for _gen_query which also returns an empty set of SqlBindParameters"""
-        query = cls._gen_query(data_source=data_source, id=id, columns=columns)
-        return (query, SqlBindParameters())
+        rendered_plan = sql_client.sql_engine_attributes.sql_query_plan_renderer.render_sql_query_plan(sql_plan)
+        return (rendered_plan.sql, rendered_plan.execution_parameters)
 
     @classmethod
-    def gen_data_source_tasks(cls, model: UserConfiguredModel) -> List[DataWarehouseValidationTask]:
+    def gen_data_source_tasks(
+        cls, model: UserConfiguredModel, sql_client: SqlClient, system_schema: str
+    ) -> List[DataWarehouseValidationTask]:
         """Generates a list of tasks for validating the data sources of the model"""
+
+        # we need a dimension to query that we know exists (i.e. the dimension
+        # is guaranteed to not cause a problem) on each data source.
+        # Additionally, we don't want to modify the original model, so we
+        # first make a deep copy of it
+        model = deepcopy(model)
+        for data_source in model.data_sources:
+            data_source.dimensions = list(data_source.dimensions) + [
+                Dimension(name=f"validation_dim_for_{data_source.name}", type=DimensionType.CATEGORICAL, expr="1")
+            ]
+
+        render_tools = QueryRenderingTools(model=model, system_schema=system_schema)
+
         tasks: List[DataWarehouseValidationTask] = []
-        for index, data_source in enumerate(model.data_sources):
+        for data_source in model.data_sources:
+            source_node = cls._data_source_node(render_tools=render_tools, data_source=data_source)
+            spec = DimensionSpec.from_name(name=f"validation_dim_for_{data_source.name}")
+            filter_elements_node = FilterElementsNode(parent_node=source_node, include_specs=[spec])
```

Comment:
Ok, so this is doing `SELECT 1 FROM {data_source.sql_query or data_source.sql_table}` for every data source, which essentially says "yes, we can execute against that rendered inner query".